ci/generate-and-upload-build-steps.sh

```diff
@@ -1,15 +1,73 @@
 #!/bin/bash
 set -euo pipefail
 
+upload_build_configuration_step() {
+    cat "ci/gdk_build.template.steps.yaml" | \
+    sed "s|ENGINE_COMMIT_HASH_PLACEHOLDER|$1|g" | \
+    sed "s|BUILD_PLATFORM_PLACEHOLDER|$2|g" | \
+    sed "s|BUILD_TARGET_PLACEHOLDER|$3|g" | \
+    sed "s|BUILD_STATE_PLACEHOLDER|$4|g" | \
+    buildkite-agent pipeline upload
+}
+
+generate_build_configuration_steps () {
+    # See https://docs.unrealengine.com/en-US/Programming/Development/BuildConfigurations/index.html for possible configurations 
+
+    # if BUILD_ALL_CONFIGURATIONS environment variable exists AND is equal to "true", then...
+    if [[ "${BUILD_ALL_CONFIGURATIONS:-false}" = "true" ]]; then
```

Comment:
Why not just trigger on the presence of this environment variable? Since we are treating it as a flag

```
	if [[ -z "${BUILD_ALL_CONFIGURATIONS} ]]; then
```

Reply:
Good point! `[[ -z "${BUILD_ALL_CONFIGURATIONS} ]]` breaks if the variable doesn't exist. I used a slight alteration `[[ -z "${BUILD_ALL_CONFIGURATIONS+x} ]]` where an empty variable is created if it doesn't exist, or the variable is assigned the value `x` if it _does_ exist.